charms/istio-pilot/lib/charms/istio_pilot/v0/istio_gateway_info_receiver.py

```diff
@@ -0,0 +1,55 @@
+import logging
+from ops.framework import Object
+from ops.model import Application
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+DEFAULT_RELATION_INTERFACE = "gateway"
+
+
+class GatewayConsumer(Object):
+    def __init__(self, charm, relation_name: str = DEFAULT_RELATION_INTERFACE):
+        super().__init__(charm, relation_name)
+        self.charm = charm
+        self.log = logging.getLogger(__name__)
+        self.relation_name = relation_name
+
+    def get_relation_data(self):
+        if not self.model.unit.is_leader():
+            return
+        gateway = self.model.relations[self.relation_name]
+        if len(gateway) == 0:
+            raise Exception("Missing gateway relation with istio-pilot")
```

Comment:
I would like to see a custom exception instead of just an `Exception` this makes it basically impossible to filter during a `try ... except` statement.